src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -11,17 +12,23 @@ namespace Stratis.Bitcoin.IntegrationTests.Common
 {
     public class TestHelper
     {
-        public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reason", int millisecondsTimeout = 50, CancellationToken cancellationToken = default(CancellationToken))
+        static async Task AsyncTaskDelay(int milisec)
+        {
+            await Task.Delay(milisec).ConfigureAwait(false);
+        }
+
+        public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reason", int millisecondsTimeout = 150, CancellationToken cancellationToken = default(CancellationToken))
         {
             cancellationToken = cancellationToken == default(CancellationToken)
-                ? new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 30 * 1000).Token
+                ? new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 40 * 1000).Token
```

Comment:
Can you give a bit of detail as to how you arrived at this particular choice of timeout? I also find that tests seem to be more reliable the longer this interval is, but perhaps the underlying tests should be corrected instead? Do we know which ones are particularly problematic?

Reply:
we managed to isolate around 12 tests that cause problems if executed in integration test suite. I think given large peformance profile of the test its just giving extra 10 seconds for the loop to spin, additionally I decreased the frequency at which we invoke methods